aten/src/ATen/core/Macros.h

```diff
@@ -9,17 +9,20 @@
 
 #ifdef _WIN32
 #if !defined(AT_CORE_STATIC_WINDOWS)
-#if defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
+// TODO: unfiy the controlling macros.
+#if defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
 #define AT_CORE_API __declspec(dllexport)
-#else
+#else // defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
 #define AT_CORE_API __declspec(dllimport)
-#endif
-#else
+#endif // defined(CAFFE2_BUILD_MAIN_LIBS) || defined(ATen_cpu_EXPORTS) || defined(caffe2_EXPORTS)
+#else // !defined(AT_CORE_STATIC_WINDOWS)
 #define AT_CORE_API
-#endif
-#else
-#define AT_CORE_API
-#endif
+#endif // !defined(AT_CORE_STATIC_WINDOWS)
+#else  // _WIN32
+#if defined(__GNUC__)
```

Comment:
Do we need to do the GNU version checks and all that business like

https://github.com/pytorch/pytorch/blob/master/caffe2/core/common.h#L117-L121

? If not, we should likely simplify both.

Reply:
I found that it seems to be working across all our builds, so I am basically only testing gnuc but not version.